### ISSUE TITLE
[BUG] Wrong work of get_it in release/profile mode flutter web

### ISSUE THREAD

**Issue by bohdan1krokhmaliuk** [fluttercommunity/get_it#245]
Having such issue in my main project. I see no idea how to fix it and it looks like get_it issue. 
Code works fine if it runs in debug mode but as soon as we run it in profile or production - it simply crashes.

I found other people who have same issues with get_it so probably it is critical. Thanks for help!

**Minimum reproducible code**:  https://github.com/bohdan1krokhmaliuk/get_it_issue.git
To reproduce:
* open project
* flutter build web --profile --dart-define=Dart2jsOptimization=O0
* cd build/web
* python3 -m http.server 8000
* checkout [localhost:8000](http://localhost:8000/#/)

![image](https://user-images.githubusercontent.com/59435984/149980066-65441be0-fbbd-4c6e-8ad1-b82ff796e252.png)

**Comment by bohdan1krokhmaliuk** [comment]
@escamoteur This problem is connected somehow to the way it is used. In case I pass GetIt.instance as an injected dependency (via constructor to my custom ServiceLocator class as example) - it emits this error. In case I use it as static GetIt.instance instead of an injected dependency - it works fine on web. 
Mobile doesn't have such problem. Hope it can help

**Comment by HKop** [comment]
Hi, I created a simple project that reproduces the same issue for me (HERE: https://github.com/HKop/getit-issue) It started happening once we extracted some of the files to a "lib_common" library and wanted to use them in the main app.

Project structure: 
1. getIt_issue main project.
2. **lib_common** library in getIt_issue folder that has components used in the main project.

Note: Issue is only present on **web** in **release** or **profile** mode. 

1. To generate getIt files - "flutter packages pub run build_runner build --delete-conflicting-outputs" in the root getIt_issue folder;
2. Then to run the app on web in profile mode - "flutter run --profile -d chrome --dart-define=Dart2jsOptimization=O0"

When opening the console, the same error is shown, yet works properly in debug mode.
